### PR TITLE
Android camera feature enhancements

### PR DESCRIPTION
**Pull request by warped-rudi**
Closes https://github.com/opencv/opencv/issues/24687

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/videoio/src/cap_android_camera.cpp [hunk 1]
```diff
@@ -478,29 +515,31 @@ class AndroidCameraCapture : public IVideoCapture
...
-                    sensitivity = (int32_t)value;
+                    sensitivity = sensitivityRange.clamp(static_cast<int32_t>(value));
                     LOGI("Setting CAP_PROP_ISO_SPEED will have no effect unless CAP_PROP_AUTO_EXPOSURE is off");
-                    camera_status_t status = ACaptureRequest_setEntry_i32(captureRequest.get(), ACAMERA_SENSOR_SENSITIVITY, 1, &sensitivity);
-                    return status == ACAMERA_OK;
+                    return submitRequest(ACaptureRequest_setEntry_i32, ACAMERA_SENSOR_SENSITIVITY, sensitivity);
                 }
                 return false;
+            case CAP_PROP_ANDROID_DEVICE_TORCH:
+                flashMode = (value != 0) ? ACAMERA_FLASH_MODE_TORCH : ACAMERA_FLASH_MODE_OFF;
```
Comment (asmorkalov) [inline modules/videoio/src/cap_android_camera.cpp:538]: There are complains about Samsung devices support here: https://stackoverflow.com/questions/23336907/what-is-the-difference-between-flash-mode-torch-and-flash-mode-on. Could you take a look?

File: modules/videoio/src/cap_android_camera.cpp [hunk 2]
```diff
@@ -732,6 +766,18 @@ class AndroidCameraCapture : public IVideoCapture
...
     }
+
+    template<typename FuncT, typename T>
+    bool submitRequest(FuncT setFn, uint32_t tag, const T &data)
+    {
+        ACaptureRequest *request = captureRequest.get();
+
+        return request &&
+               setFn(request, tag, 1, &data) == ACAMERA_OK &&
+               ACameraCaptureSession_setRepeatingRequest(captureSession.get(),
```
Comment (asmorkalov) [inline modules/videoio/src/cap_android_camera.cpp:777]: Why do we need it for every ACaptureRequest_setEntryXXX call?

File: modules/videoio/src/cap_android_camera.cpp [hunk 3]
```diff
@@ -561,7 +600,7 @@ class AndroidCameraCapture : public IVideoCapture
             return false;
         }
         std::shared_ptr<ACameraMetadata> cameraMetadata = std::shared_ptr<ACameraMetadata>(metadata, deleter_ACameraMetadata);
-        ACameraMetadata_const_entry entry;
+        ACameraMetadata_const_entry entry = {};
```
Comment (asmorkalov) [inline modules/videoio/src/cap_android_camera.cpp:603]: The  change does nothing. Default constructor was called in old version too.

File: modules/videoio/src/cap_android_camera.cpp (hunk 1 above)
Comment (warped-rudi) [reply modules/videoio/src/cap_android_camera.cpp:538]: Hmm, I don't know. My primary test device is a SONY. I will try, if FLASH_MODE_ON will work as well.

File: modules/videoio/src/cap_android_camera.cpp (hunk 2 above)
Comment (warped-rudi) [reply modules/videoio/src/cap_android_camera.cpp:777]: Yes. It is not sufficient to update the metadata only. Re-submitting the capture request will make the change effective after the current request has finished.

File: modules/videoio/src/cap_android_camera.cpp (hunk 1 above)
Comment (asmorkalov) [reply modules/videoio/src/cap_android_camera.cpp:538]: Just tested torch with Xiaomi Mi 10 and Samsung Galaxy S10. Both works well. Let's ignore it for now.

**Review by asmorkalov — APPROVED** [review]

File: modules/videoio/src/cap_android_camera.cpp (hunk 1 above)
Comment (warped-rudi) [reply modules/videoio/src/cap_android_camera.cpp:538]: According to https://developer.android.com/reference/android/hardware/Camera.Parameters  FLASH_MODE_ON is part of the old API. Camera2 has FLASH_MODE_OFF, FLASH_MODE_SINGLE and FLASH_MODE_TORCH only.

**Comment by warped-rudi** [comment]
Thanks! I have more in the pipe, including zoom support...

**Pull request merged**